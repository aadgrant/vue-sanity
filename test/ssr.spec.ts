import Vue from 'vue'
import VueCompositionApi from '@vue/composition-api'
import {
  createBundleRenderer,
  BundleRendererOptions,
  BundleRenderer,
} from 'vue-server-renderer'
import path from 'path'

import { compileWithWebpack } from './helpers/webpack'

Vue.use(VueCompositionApi)
Vue.config.productionTip = false
Vue.config.devtools = false

jest.setTimeout(10000)

function createRenderer(
  file: string,
  options: BundleRendererOptions,
  cb: (renderer: BundleRenderer) => void
) {
  compileWithWebpack(
    file,
    {
      target: 'node',
      devtool: false,
      output: {
        path: path.resolve(__dirname, './dist'),
        filename: 'bundle.js',
        libraryTarget: 'umd',
      },
    },
    fs => {
      const bundle = fs.readFileSync(
        path.resolve(__dirname, './dist/bundle.js'),
        'utf-8'
      )
      const renderer = createBundleRenderer(bundle, options)
      cb(renderer)
    }
  )
}

describe('ssr', () => {
  // eslint-disable-next-line
  test('cache fetches data correctly on SSR', async done => {
    createRenderer('app.ts', {}, renderer => {
      renderer.renderToString({}, (err, res) => {
        expect(err).toBeNull()
        expect(res).toMatchSnapshot()
        done()
      })
    })
  })
})
