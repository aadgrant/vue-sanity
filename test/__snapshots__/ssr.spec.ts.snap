// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr cache fetches data correctly on SSR 1`] = `"<div data-server-rendered=\\"true\\">data: value, status: server loaded</div>"`;
